backend/compact-connect/lambdas/provider-data-v1/tests/function/test_data_model/test_provider_transformations.py

```diff
@@ -0,0 +1,175 @@
+import json
+from unittest.mock import patch
+
+from boto3.dynamodb.conditions import Key
+from botocore.exceptions import ClientError
+from moto import mock_aws
+from tests.function import TstFunction
+
+
+@mock_aws
+class TestTransformations(TstFunction):
+    # Yes, this is an excessively long method. We're going with it for sake of a single illustrative test.
+    def test_transformations(self):  # pylint: disable=too-many-statements,too-many-locals
+        """
+        Provider data undergoes several transformations from when a license is first posted, stored into the database,
+        then returned via the API. We will specifically test that chain, end to end, to make sure the transformations
+        all happen as expected.
+        """
+        # Before we get started, we'll pre-set the SSN/providerId association we expect
+        with open('tests/resources/dynamo/provider-ssn.json', 'r') as f:
+            provider_ssn = json.load(f)
+
+        self._table.put_item(
+            Item=provider_ssn
+        )
```

Comment:
Ordinarily it would, but I'm cheating a bit here. If we only created the provider via the API, it would be assigned a new UUID, which I couldn't predict, making the rest of this test more difficult. Instead, I'm forcing a known outcome by creating the provider-ssn association record in advance, with the UUID I already know, so that I can test the remaining data transformations against the UUID I expect to have.

Reply:
Makes sense!